packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -0,0 +1,45 @@
+import { attr } from '@microsoft/fast-element';
+import { DesignSystem, FoundationElement } from '@microsoft/fast-foundation';
+import { template } from '../theme-provider/template';
+import type { WaferMapRenderingObject } from './models/wafer-map-rendering-object';
+import { styles } from './styles';
+import { WaferMapColorBy, WaferMapColorsScale } from './types';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-wafer-map': WaferMap;
+    }
+}
+
+/**
+ * A nimble-styled WaferMap
+ */
+export class WaferMap extends FoundationElement {
+    @attr({
+        attribute: 'waferData'
+    })
+    public waferData!: WaferMapRenderingObject;
+
+    @attr({
+        attribute: 'colorBy'
+    })
+    public colorBy: WaferMapColorBy = WaferMapColorBy.hardBin;
+
+    @attr({
+        attribute: 'highlightedValues'
+    })
+    public highlightedValues!: string[];
+
+    @attr({
+        attribute: 'colorsScale'
+    })
+    public colorsScale!: WaferMapColorsScale;
```

Comment:
Aren't all TS class properties public by default ? I am wondering if there is a convention that we are following or if there is something else - the reason why I am asking this, is that in all of the FAST examples and documentation the "public" keyword is omitted. I did see however that everywhere in nimble we do explicitly specify attributes as "public" ... it does feel a little bit like an overhead :)

Reply:
This repo (and hopefully all of NI's JS/TS code) uses the [NI ESLint styleguide](https://github.com/ni/javascript-styleguide) and it [configures the `explicit-member-accessibility` rule](https://github.com/ni/javascript-styleguide/blame/726576b5883b5bb618201987f88c86e9636eef83/packages/eslint-config-typescript/index.js#LL79C21-L79C21) to require the `public` keyword. 

The styleguide team had a long debate about whether to require the `public` keyword but ended up deciding to enforce it because it's more explicit and NI has a lot of developers coming from the C# world so it is helpful to prevent confusion if devs don't know that the default accessibility is public for JS even though it's private for C#.